internal/incident/incidents.go

```diff
@@ -25,34 +23,116 @@ var (
 func LoadOpenIncidents(ctx context.Context, db *icingadb.DB, logger *logging.Logger, runtimeConfig *config.RuntimeConfig) error {
 	logger.Info("Loading all active incidents from database")
 
-	var objectIDs []types.Binary
-	err := db.SelectContext(ctx, &objectIDs, `SELECT object_id FROM incident WHERE "recovered_at" IS NULL`)
-	if err != nil {
-		logger.Errorw("Failed to load active incidents from database", zap.Error(err))
+	currentIncidentsMu.Lock()
+	defer currentIncidentsMu.Unlock()
 
-		return errors.New("failed to fetch open incidents")
-	}
+	incidents := make(chan *Incident, 1)
+	g, childCtx := errgroup.WithContext(ctx)
+
+	g.Go(func() error {
+		defer close(incidents)
 
-	for _, objectID := range objectIDs {
-		obj, err := object.LoadFromDB(ctx, db, objectID)
+		query := `
+			SELECT incident.id, started_at, severity, incident.object_id, object.source_id, object.name, object.url, oit.tag, oit.value, oet.tag, oet.value
+			FROM incident
+				INNER JOIN object on object.id = incident.object_id
+				INNER JOIN public.object_id_tag oit on object.id = oit.object_id
+				LEFT JOIN public.object_extra_tag oet on object.id = oet.object_id
+			WHERE "recovered_at" IS NULL
+			ORDER BY incident.id, object.id`
```

Comment:
This will select quite a lot of redundant information. Especially with lots of tags, you get the Cartesian product of ID tags and extra tags, i.e. `len(id_tags) * max(1, len(extra_tags))` rows. We don't have a limit on the number of tags (note in particular that we're not limited to `host` + `service`), so this can get large quite quickly.

So it might be worth it to split this across multiple queries. This would then also allow to get some useful parallelism here. Like do an query like this first:

```sql
SELECT id, object_id, [...] FROM incident WHERE recovered_at IS NULL;
```

And then take the resulting object IDs and perform queries like this with them:

```sql
SELECT [...] FROM object WHERE id IN (?, ?, [...]);
SELECT [...] FROM object_id_tag WHERE object_id IN (?, ?, [...]);
SELECT [...] FROM object_extra_tag WHERE object_id IN (?, ?, [...]);
```

Where those queries could be run in parallel. Also, the set of object IDs has to be partitioned to avoid running into number of placeholders limits, which happens to allow additional parallelism. But don't overdo it, we don't want to overwhelm the database and also keep the code complexity reasonable. So simply partitioning the object IDs and running the three queries in sequence for each subset might be good enough, as this should save a lot of synchronization when merging the results and if there's enough data to be read that actually makes parallelization relevant, enough of it should already happen due to processing different IDs in parallel.